docs/models/naming-strategies.mdx

```diff
@@ -0,0 +1,314 @@
+
+```javascript
+import { Model, Sequelize } from '@sequelize/core';
+
+class User extends Model {}
+
+const sequelize = new Sequelize({
+  // ...
+  models: [User],
+});
+
+console.log(User.table.tableName);
+// → Users
+```
+
+### Snake-case table names
+
+By default, Sequelize does not change the casing of your model name when inferring the table name.
+
+In some databases, it is conventional to have table names in [snake_case](https://en.wikipedia.org/wiki/Snake_case).
+You can make Sequelize generate a snake_cased table name by setting the [`underscored`](pathname:///api/interfaces/ModelOptions.html#underscored) option.
+
+:::info
+
+The `underscored` option also impacts Column Names. [See *Snake-case column names*](#snake-case-column-names).
+
+:::
+
+```javascript
+import { Model, Sequelize } from '@sequelize/core';
+import { Table } from '@sequelize/core/decorators-legacy';
+
+@Table({
+  // highlight-next-line
+  underscored: true,
+})
+class User extends Model {}
+
+const sequelize = new Sequelize('sqlite::memory:', {
+  models: [User],
+});
+
+console.log(User.table.tableName);
+// → users (notice the lowercase u)
+```
+
+This behavior can also be defined globally by setting the option in the Sequelize constructor:
+
+```js
+import { Model, Sequelize } from '@sequelize/core';
+
+class User extends Model {}
+
+const sequelize = new Sequelize('sqlite::memory:', {
+  define: {
+    // highlight-next-line
+    underscored: true,
+  },
+  models: [User],
+});
+
+console.log(User.table.tableName);
+// → users
+```
+
+### Enforcing the table name to be equal to the model name
+
+You can stop the auto-pluralization performed by Sequelize using the [`freezeTableName: true`](pathname:///api/interfaces/ModelOptions.html#freezeTableName) option.
+Sequelize will use the model name as the table name, without any modifications.
+
+Note that setting `freezeTableName` also causes Sequelize to ignore the `underscored` option for table names.
+
+```javascript
+import { Model, Sequelize } from '@sequelize/core';
+import { Table } from '@sequelize/core/decorators-legacy';
+
+@Table({
+  // highlight-next-line
+  freezeTableName: true,
+})
+class User extends Model {}
+
+const sequelize = new Sequelize('sqlite::memory:', {
+  models: [User],
+});
+
+console.log(User.table.tableName);
+// → User
+```
+
+This behavior can also be defined globally by setting the option in the Sequelize constructor:
+
+```js
+import { Model, Sequelize } from '@sequelize/core';
+
+class User extends Model {}
+
+const sequelize = new Sequelize('sqlite::memory:', {
+  define: {
+    // highlight-next-line
+    freezeTableName: true,
+  },
+  models: [User],
+});
+
+console.log(User.table.tableName);
+// → User
+```
+
+### Manually setting the table name
+
+You can simply tell Sequelize the name of the table directly as well.
+This bypasses any automatic table name generation option. Neither snake_casing nor pluralization will be done:
```

Comment:
```suggestion
You can also tell Sequelize the name of the table directly. This will skip any automatic table name generation options and bypass snake_casing and pluralization.
```